workshop/4-Messaging/4.1-Pinpoint.ipynb

```diff
@@ -817,6 +817,172 @@
+    "\n",
+    "1. On the SMS and voice settings page, under Number settings, choose the long code that you received in the previous section.\n",
+    "2. Under Two-Way SMS, choose Enable two-way SMS.\n",
+    "3. Under **Incoming message destination**, choose **Choose an existing SNS topic**. From the drop down menu, choose the SNS topic that has name `PinpointIncomingTextAlertsSNSTopic` in its ARN.\n",
+    "4. Under Two-way SMS keywords, for Keyword, enter the word that customers send you to confirm their subscriptions. In this case, enter **Y** as the keyword.\n",
+    "5. Under Response Message, enter `Retail Demo Store: Use code 839405826 for an extra 20% off.` (Since we want to send the offer code to the user once they confirm their subscription by replying with \"Y\" to the confirmation message.)\n",
+    "\n",
+    "![Pinpoint Two Way SMS](images/pinpoint-two-way-sms-enable.png)\n"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "source": [
+    "<div class=\"alert alert-info\">\n",
+    "<strong>This section assumes that you're leasing a long code that's based in the United States or Canada. If you follow the procedures in this section, but choose a country other than the United States or Canada, you won't be able to use that number to send SMS messages. To learn more about leasing SMS-capable long codes in countries other than the United States or Canada, see <span href=\"https://docs.aws.amazon.com/pinpoint/latest/userguide/channels-sms-awssupport-long-code.html\"> Requesting dedicated long codes for SMS messaging with Amazon Pinpoint in the Amazon Pinpoint User Guide. </span>\n",
+    "</strong>\n",
+    "</div>"
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "source": [
+    "## Create Retail Semo Store User Accounts & Amazon Pinpoint SMS Endpoints\n",
+    "Next, refer to the previous section of this workshop to create a new user account if you haven't done so already. Once you have signed in as a Retail Demo Store user, the text alerts section is shown near the footer of the app as shown in the figure below. \n",
+    "\n",
+    "![Text alerts section] (images/pinpoint-text-alerts-ui.png) \n",
+    "\n",
+    "* Enter your phone number along with the \"+\" and country code (without any spaces). Eg. \"+1XXXXXXXXXX\"\n",
+    "* This will send a confirmation SMS message to the phone number entered above, asking the user to confirm if they want to receive a one time SMS from Retail Demo Store. \n",
+    "* Reply with a Y in order to opt in for receiving messages. If you reply with anything apart from Y, the SMS endpoint will be created, but it will not be opted in i.e. the segment and campaigns that we create in the next steps, will not send messages to the user unless they have opted in for the messages.\n",
+    "* Once you reply with Y, the endpoint is opted in to receive messages from Pinpoint.\n",
```

Comment:
The Lambda makes this all work. Where is that described in the notebook?

Reply:
I'll add it.